src/frontegg-mock/src/lib.rs

```diff
@@ -538,7 +553,161 @@ async fn handle_post_user_api_token<'a>(
+async fn handle_list_user_api_tokens(
+    State(context): State<Arc<Context>>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+) -> Result<Json<Vec<UserApiTokenResponse>>, StatusCode> {
+    let claims = match decode_access_token(&context, authorization.token()) {
+        Ok(TokenData { claims, .. }) => claims,
+        Err(_) => return Err(StatusCode::UNAUTHORIZED),
+    };
+
+    let user_api_tokens = context.user_api_tokens.lock().unwrap();
+    let tokens: Vec<UserApiTokenResponse> = user_api_tokens
+        .iter()
+        .filter(|(_, email)| *email == claims.email.as_ref().unwrap())
+        .map(|(token, _)| UserApiTokenResponse {
+            client_id: token.client_id.to_string(),
+            description: "User API Token".to_string(),
+            created_at: Utc::now(),
+            secret: token.secret.to_string(),
+        })
+        .collect();
+
+    Ok(Json(tokens))
+}
+
+// https://docs.frontegg.com/reference/userapitokensv1controller_deleteapitoken
+async fn handle_delete_user_api_token(
+    State(context): State<Arc<Context>>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+    Path(token_id): Path<Uuid>,
+) -> StatusCode {
+    let claims = match decode_access_token(&context, authorization.token()) {
+        Ok(TokenData { claims, .. }) => claims,
+        Err(_) => return StatusCode::UNAUTHORIZED,
+    };
+
+    let mut user_api_tokens = context.user_api_tokens.lock().unwrap();
+
+    let removed = user_api_tokens
+        .iter()
+        .find(|(token, email)| {
+            token.client_id == token_id && *email == claims.email.as_ref().unwrap()
+        })
+        .map(|(token, _)| token.clone());
+
+    if let Some(token_to_remove) = removed {
+        user_api_tokens.remove(&token_to_remove);
+        StatusCode::OK
+    } else {
+        StatusCode::NOT_FOUND
+    }
+}
+
+// https://docs.frontegg.com/reference/tenantapitokensv1controller_gettenantsapitokens
+async fn handle_list_tenant_api_tokens(
+    State(context): State<Arc<Context>>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+) -> Result<Json<Vec<TenantApiTokenResponse>>, StatusCode> {
+    let _claims = match decode_access_token(&context, authorization.token()) {
+        Ok(TokenData { claims, .. }) => claims,
+        Err(_) => return Err(StatusCode::UNAUTHORIZED),
+    };
+
+    let tenant_api_tokens = context.tenant_api_tokens.lock().unwrap();
+    let tokens: Vec<TenantApiTokenResponse> = tenant_api_tokens
+        .iter()
+        .map(|(api_token, config)| TenantApiTokenResponse {
+            client_id: api_token.client_id,
+            description: "".to_string(),
+            secret: api_token.secret,
+            created_by_user_id: Uuid::nil(),
+            metadata: config
+                .metadata
+                .as_ref()
+                .and_then(|m| m.user.clone().map(serde_json::Value::String)),
+            created_at: Utc::now(),
```

Comment:
As above, we should probably store the description, created_by_user_id and created_at.